packages/graphile-build-pg/src/plugins/viaTemporaryTable.js

```diff
@@ -43,7 +43,11 @@ export default async function viaTemporaryTable(
   sqlMutationQuery: SQL,
   sqlResultSourceAlias: SQL,
   sqlResultQuery: SQL,
-  isPgClassLike: boolean = true
+  isPgClassLike: boolean = true,
+  isPgRecord: boolean = false,
+  // eslint-disable-next-line flowtype/no-weak-types
+  outputArgTypes: Array<any> = [],
+  outputArgNames: Array<string> = []
```

Comment:
Can you confirm these changes are backwards compatible? `viaTemporaryTable` is now an API that plugins use, I don't want to break any of them.

Reply:
I wonder if we should merge these things into one argument - an object - that we just use for records. That way changes can be made easier in future without risking breaking things.

Reply:
I believe the changes to viaTemporaryTable are backwards compatible. `isPgRecord` defaults to false, so scalars will just pass through to the else clause of the two ternaries I inserted. I did have to refactor `convertFieldBack` to include more of the query, but resulting SQL is identical for both classes and scalars. `outputArgTypes` and `outputArgNames` are only referenced if `isPgRecord` is passed as true, so no breakage there.

Plugins might have issues handling all the new procs they'll be seeing on introspectionResultsByKind.proc, perhaps incorrectly inferring an incorrect value for isPgClassLike and passing that through to viaTemporaryTable.. but that's not really a breaking change in the viaTemporaryTable API itself.

As far as wrapping record-related args into an object, are you suggesting `{ outputArgTypes: Array<any> = [],
  outputArgNames: Array<string> = [] }`, or including isPgRecord as well? I think isPgRecord should be adjacent to isPgClassLike, but I'm fine putting the other two into an object.

Reply:
I was thinking that the presence of the `{ outputArgTypes, outputArgNames }` object would infer `isPgRecord` - send null/undefined if `!isPgRecord`

Reply:
Done. afb4357